src/datasets/datasets.v4.controller.ts

```diff
@@ -778,9 +780,31 @@ Set \`content-type\` header to \`application/merge-patch+json\` if you would lik
   async findByIdAndUpdate(
     @Req() request: Request,
     @Param("pid") pid: string,
+    @Headers() headers: Record<string, string>,
     @Body()
     updateDatasetDto: PartialUpdateDatasetDto,
   ): Promise<OutputDatasetDto | null> {
+    return this.findByIdAndUpdateInternal(
+      request,
+      pid,
+      headers,
+      updateDatasetDto,
+    );
+  }
+
+  async findByIdAndUpdateInternal(
+    @Req() request: Request,
+    @Param("pid") pid: string,
+    @Headers() headers: Record<string, string>,
+    @Body()
+    updateDatasetDto: PartialUpdateDatasetDto,
+  ): Promise<OutputDatasetDto | null> {
```

Comment:
Why do we define function `findByIdAndUpdateInternal` and call it only from `findByIdAndUpdate`?
Can we combine `findByIdAndUpdate` and `findByIdAndUpdateInternal`?